mcstatus/motd.py

```diff
@@ -0,0 +1,280 @@
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing_extensions import Self
+
+__all__ = [
+    "Motd",
+    "Formatting",
+    "MinecraftColor",
+    "WebColor",
+    "RawMotd",
+]
+
+MOTD_COLORS_RE = re.compile(r"([\xA7|&][0-9A-FK-OR])", re.IGNORECASE)
+
+
+class Formatting(Enum):
+    """Enum for Formatting codes.
+
+    See `MineCraft wiki <https://minecraft.fandom.com/wiki/Formatting_codes#Formatting_codes>`_
+    for more info.
+
+    .. note::
+        ``STRIKETHROUGH`` and ``UNDERLINED`` doesn't work on Bedrock, but our parser
+        doesn't keep it in mind. See `MCPE-41729 <https://bugs.mojang.com/browse/MCPE-41729>`_.
+    """
+
+    BOLD = "l"
+    ITALIC = "o"
+    UNDERLINED = "n"
+    STRIKETHROUGH = "m"
+    OBFUSCATED = "k"
+    RESET = "r"
+
+
+class MinecraftColor(Enum):
+    """Enum for Color codes.
+
+    See `MineCraft wiki <https://minecraft.fandom.com/wiki/Formatting_codes#Color_codes>`_
+    for more info.
+    """
+
+    BLACK = "0"
+    DARK_BLUE = "1"
+    DARK_GREEN = "2"
+    DARK_AQUA = "3"
+    DARK_RED = "4"
+    DARK_PURPLE = "5"
+    GOLD = "6"
+    GRAY = "7"
+    DARK_GRAY = "8"
+    BLUE = "9"
+    GREEN = "a"
+    AQUA = "b"
+    RED = "c"
+    LIGHT_PURPLE = "d"
+    YELLOW = "e"
+    WHITE = "f"
+
+    # Only for bedrock
+    MINECOIN_GOLD = "g"
+
+
+@dataclass(kw_only=True)
+class WebColor:
+    """Raw HTML color from MOTD.
+
+    Can be found in MOTD when someone uses gradient.
+
+    .. note:: Every value here, auto generates if it's None.
+    """
+
+    hex: str = None  # type: ignore[assignment]
+    rgb: Tuple[int, int, int] = None  # type: ignore[assignment]
+
+    def __post_init__(self) -> None:
+        """Post init method.
+
+        Ensures that ``WebColor.hex`` starts with '#', and generate RGB color for it,
+        if ``WebColor.rgb`` is None (default value) or generate hex value from RGB, if
+        ``WebColor.hex`` is None (also default value).
+
+        :raises ValueError: When hex color isn't correct, so we can't generate RGB for it.
+        """
+        if self.hex is None and self.rgb is None:
+            raise ValueError("`hex` and `rgb` attributes is None, we need one of them.")
+
+        if self.hex is None:
+            if not all(256 > e > -1 for e in self.rgb):
+                raise ValueError("One of RGB colors is less than zero and more than 255: " + str(self.rgb))
+            self.hex = "#%02x%02x%02x" % self.rgb
+        else:
+            if not self.hex.startswith("#"):
+                self.hex = "#" + self.hex
+            hex = self.hex.lstrip("#")
+            try:
+                self.rgb = tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))  # type: ignore[assignment]
+            except ValueError:
+                raise ValueError(f"Incorrect hex color '{self.hex}', failed to generate RGB.")
+
+
+RawMotd = List[Union[Formatting, MinecraftColor, WebColor, str]]
+"""Type alias to field ``Motd.raw``."""
+
+
+@dataclass
+class Motd:
```

Comment:
Main reason why I named it `motd`, because there is no `description` in #306, so there will be less work on merge conflict when I will port this change to #306.

I would prefer discuss it in #306, because there is already open conversation about it.

Reply:
This PR is asking for a merge into our master branch, not into the branch of 306, that PR may take time to get merged, and for now, we should keep things consistent with what's currently in the master branch. If you absolutely do not wish to do that, because it would be too much work to change it back later (though that chaging back would probably happen in 306 since this will likely get merged before that), mark this PR as draft (or let me know, if it can still be reviewed but just shouldn't be merged,ad I can just mark it with do-not-merge label) and make a comment in PR's description explaining that it's blocked by 306.

Reply:
Then you should propose deprecation variants, as I don't know anything better than rename the field. Or I can just don't do any deprecations?

Reply:
I removed all deprecations and almost all changes, except for `motd.py` and `test_motd.py`, [here](https://github.com/py-mine/mcstatus/compare/23470e815b2c730cf567111950cec7404b71a462..fd46d485eceb135815aa033c17edd070d0acf398) and [here](https://github.com/py-mine/mcstatus/compare/27021e6d8995c52d2bed11420daf48e6ec03d5a9..3d72f22bfdd514469de253cb2222c6aa66e343f3). So this PR will not produce such many merge conflicts.

I will remove those, when I will rebase this PR on top of #306, as we discussed [somewhere here](https://github.com/py-mine/mcstatus/pull/306/files/ec77b344c6ca59bf16f8667da49d3decb6f2276d#r985899725).